hack/build-release.sh

```diff
@@ -8,8 +8,11 @@ mkdir -p .imgpkg/
 # makes the get_kappctrl_ver function available (scrapes version from git tag)
 source $(dirname "$0")/version-util.sh
 
-ytt -f config/ -f config-release -v kapp_controller_version="$(get_kappctrl_ver)" --data-values-env=KCTRL | kbld --imgpkg-lock-output .imgpkg/images.yml -f- > ./tmp/release.yml
+export version="$(get_kappctrl_ver)"
 
+yq eval '.metadata.annotations."kapp-controller.carvel.dev/version" = env(version)' -i "config/deployment.yml"
+
+ytt -f config/ -f config-release -v dev.kapp_controller_version="$(get_kappctrl_ver)" --data-values-env=KCTRL | kbld --imgpkg-lock-output .imgpkg/images.yml -f- > ./tmp/release.yml
```

Comment:
I'm still trying to figure out why secretgen-controller needed the `v` prefix to be removed but KC doesn't:

https://github.com/vmware-tanzu/carvel-secretgen-controller/blob/develop/hack/build-and-publish-release.sh#L15

Reply:
Not sure about this part. Should I do the same here as well?